presto-main/src/main/java/io/prestosql/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -52,6 +55,8 @@
 public class IterativeOptimizer
         implements PlanOptimizer
 {
+    private static final boolean SHUFFLE_RULES = Boolean.valueOf(System.getProperty("io.prestosql.sql.planner.iterative.IterativeOptimizer.SHUFFLE_RULES", "false"));
```

Comment:
Let's not do that :)

I'd rather have a feature config flag that can be set during tests. One issue this raises is whether tests are reproducible and how one would go about tracking down a problem caused by a specific ordering of rule application.

Reply:
@martint i understand the drawbacks.
OTOH, Travis results themselves speak for this PR.

I can do this as a feature config flag, but I would still want to have it enabled during tests execution.

For tests reproducibility, we could do this:
- choose seed randomly, print it
- reshuffle using the seed

Then, when Travis job fails, the only thing you need to repro is this seed number.

Reply:
Anyway, i am open for suggestions.

Reply:
Maybe we could have a config property to set the seed? Then seed would be printed out of the box.

Reply:
@martint WDYT?

Reply:
That's a good idea.

Reply:
@martint @kokosing so i technical question -- tests are executed in parallel, so if i pick random seed in each test class independently, I will not know which seed was used for a failing test.
For this reason, I'd prefer to choose one seed and save it on a static field.
So, the configuration toggle would be `experimenta.iterative-optimizer.shuffle-rules = true | false`.
The seed would be chosen and printed the first time such Iterative Optimizer is instantiated. 
Are you ok with this?

Reply:
Maybe you could inject seed with a system property, so each test class would have same seed?

Reply:
@kokosing do you know how can i inject a randomly-chosen seed from maven (pom) level? Or did you have something else on your mind?

Reply:
See https://maven.apache.org/surefire/maven-surefire-plugin/examples/system-properties.html#. 

Environment variables can be referenced using the `env` prefix, like `${env.M2_HOME}`.